### DISCUSSION TITLE
Prisma with CosmosDB, MongoDB connector have error with sharded collection

### DISCUSSION THREAD

**Discussion by dohaicuong** [prisma/prisma#9544] [Q&A] [answered]
Hi Prisma Team! I just got this error when setup Prisma with MongoDB on Azure CosmosDB and create a sharded collection

```
Invalid `prisma.user.create()` invocation:\n\n\n  Error occurred during query execution:\nConnectorError(ConnectorError {
user_facing_error: None, kind: RawError { 
code: \"unknown\", message: \"Command failed (50841): Multi-document transactions cannot be run in a sharded collection)\" 
} })
```

# Versions

- Node: v14.17.4
- OS: macOs
- Prisma client: 3.1.1
- Database: MongoDB 4.0 on Azure CosmosDB

# Schema
```prisma
datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["mongoDb"]
}

model User {
  id String @id @default(dbgenerated()) @map("_id") @db.ObjectId
  name String
}
```

# MongoDB setup

- Create mongoDB 4.0 on cosmosDB
- Create new DB and User collection with shard key

# Connection string
```
mongodb://user:pass@url/db_name?ssl=true
&connectTimeoutMS=5000
&maxPoolSize=50
&replicaSet=globaldb
&retrywrites=false
&maxIdleTimeMS=120000
&appName=@app_name@
```

# Prisma command
```ts
const newUser = await prisma.user.create({
  data: { name: 'name' }
})
```

# Errors
```
Invalid `prisma.user.create()` invocation:\n\n\n  Error occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: None, kind: RawError { code: \"unknown\", message: \"Command failed (50841): Multi-document transactions cannot be run in a sharded collection)\" } })
```

**Comment by janpio** [comment, marked as answer]
Looks like something that is not supported yet. Can you turn this into an issue so we can track it in our reproduction and development process? Thanks.

**Reply by janpio** [reply]
Issue was created here: https://github.com/prisma/prisma/issues/9566

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
